PoSSapi/src/Application/Orders/Commands/UpdateOrderCommand.cs

```diff
@@ -32,12 +32,15 @@ public async Task<Unit> Handle(UpdateOrderCommand request, CancellationToken can
     {
         var order = await _context.Orders.FindAsync(request.Id, cancellationToken)
                     ?? throw new NotFoundException(nameof(Order), request.Id);
+        
+        var customer = await _context.Users.FindAsync(request.CustomerId, cancellationToken)
+                        ?? throw new NotFoundException(nameof(User), request.CustomerId);
+
+        _context.OrderedDishes.Where(d => d.OrderId == request.Id).ToList()
+            .ForEach(d => _context.OrderedDishes.Remove(d));
```

Comment:
Doesn't this depend on the status change? What if by updating the user wanted to add another dish to the order?